test/hotspot/jtreg/serviceability/sa/ClhsdbSymbol.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.apps.LingeredApp;
+import jtreg.SkippedException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @test
+ * @bug 8261095
+ * @summary Test the clhsdb 'symbol' command on live process
+ * @requires vm.hasSA
+ * @library /test/lib
+ * @run main/othervm ClhsdbSymbol
+ */
+
+public class ClhsdbSymbol {
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("Starting the ClhsdbSymbol test");
+        LingeredApp theApp = null;
+
+        try {
+            List<String> cmds = null;
+            String cmdStr = null;
+            Map<String, List<String>> expStrMap = null;
+            ClhsdbLauncher test = new ClhsdbLauncher();
+            theApp = LingeredApp.startApp();
+            System.out.println("Started LingeredApp with pid " + theApp.getPid());
+
+            // Use command "class java.lang.Thread" to get the address of the InstanceKlass for java.lang.Thread
+            cmdStr = "class java.lang.Thread";
+            cmds = List.of(cmdStr);
+            expStrMap = new HashMap<>();
+            expStrMap.put(cmdStr, List.of("java/lang/Thread"));
+            String classOutput = test.run(theApp.getPid(), cmds, expStrMap, null);
+
+            // Extract the thread InstanceKlass address from the output, which looks similar to the following:
+            //   java/lang/Thread @0x000000080001d940
+            String threadAddress = classOutput.lines().filter(part -> part.startsWith("java/lang/Thread"))
+                                                      .map(part -> part.split(" @"))
+                                                      .findFirst()
+                                                      .map(addresses -> addresses[1])
+                                                      .orElseThrow(() -> new RuntimeException("Cannot find address of " +
+                                                              "the InstanceKlass for java.lang.Thread in output"));
```

Comment:
These lines really don't format well, a reason why the previous `orElse()` version is probably better. Maybe you can make this work by moving the `.filter()` call to a new line that is indented 8 more than the previous line

Reply:
After moving the filter to the next line also can not fit this in one line.
In addition to the filter, moving all characters starting from "new RuntimeException(...." also goes till 126 char.
Following is one suggestion, does it look good?

String threadAddress = classOutput.lines()
        .filter(part -> part.startsWith("java/lang/Thread"))
        .map(part -> part.split(" @"))
        .findFirst()
        .map(addresses -> addresses[1])
        .orElseThrow(() -> new RuntimeException(
                "Cannot find address of the InstanceKlass for java.lang.Thread in output"));

Reply:
Formatting in my comment is not clear, so I have added one new commit for this.

Reply:
Formatting looked fine in the email. Try putting it in a code block to prevent the auto formatting being done on it.

Latest webrev looks good.